### PR TITLE
[13.0][MIG+IMP] users_ldap_groups: Migration to 13.0

### PR DESCRIPTION
**Pull request by joao-p-marques**
Standard migration + Improvement to warn user when configuring Odoo to override the default groups with default ones. One of the groups should be a "User type", otherwise the user won't be correctly created.

Also affected by https://github.com/odoo/odoo/issues/41060, but it is a non-blocking issue. 

@Tecnativa 
TT28153

@Yajo @pedrobaeza can you review please?

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@Yajo can you please check?

File: users_ldap_groups/models/res_company_ldap.py [hunk 1]
```diff
@@ -48,5 +55,16 @@ def _get_or_create_user(self, conf, login, ldap_entry):
...
                 )
                 groups.append((4, mapping.group_id.id, False))
+        if (
+            this.only_ldap_groups
+            and len([g[1] for g in groups if g[0] == 4 and g[1] in essential_groups])
+            != 1
+        ):
+            raise UserError(
+                _(
+                    "The created user needs to have one (and only one) of the"
```
Comment (pedrobaeza) [inline users_ldap_groups/models/res_company_ldap.py:65]: Why do you need this check? Odoo already assures this.

**Review by pedrobaeza — APPROVED** [review]
Except my comment

File: users_ldap_groups/models/res_company_ldap.py (hunk 1 above)
Comment (joao-p-marques) [reply users_ldap_groups/models/res_company_ldap.py:65]: Not exactly. This issue happens when we select the `only_ldap_groups` option, thus replacing all of the default groups added by Odoo by the ones we define. 
If we don't set one (and only one) of the user types, we will get an inconsistent behavior without a proper warning by Odoo, so IMHO it is better to give this error to the user and stop the creation of the new user.

**Review by pedrobaeza — APPROVED** [review]

File: users_ldap_groups/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2012-2018 Therp BV <https://therp.nl>
+# Copyright 2018 Brainbean Apps <https://brainbeanapps.com>
+# Copyright 2021 Tecnativa - João Marques
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+{
+    "name": "LDAP groups assignment",
+    "version": "13.0.1.0.0",
+    "depends": ["auth_ldap"],
+    "author": "Therp BV, " "Odoo Community Association (OCA)",
```
Comment (primes2h) [inline users_ldap_groups/__manifest__.py:10]: Repo website is missing.

File: users_ldap_groups/models/res_company_ldap_operator.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+from odoo import api, models
+
+_logger = getLogger(__name__)
+
+
+class ResCompanyLdapOperator(models.AbstractModel):
+    """Define operators for group mappings"""
+
+    _name = "res.company.ldap.operator"
+    _description = "Definition op LDAP operations"
```
Comment (primes2h) [inline users_ldap_groups/models/res_company_ldap_operator.py]:
```suggestion
    _description = "Definition of LDAP operations"
```

**Review by primes2h — CHANGES_REQUESTED** [review]

**Comment by joao-p-marques** [comment]
Thanks @primes2h changes made!

**Review by primes2h — APPROVED** [review]
:+1:

**Pull request merged**